### PR TITLE
Features/Angular Pretty URLs

### PR DESCRIPTION
**Pull request by kaptron**
Was something Jay Nath requested, and it seemed easy enough to give it a shot. (only took 5 mins)

Played with setting up pretty URLs in angular ui-router, seems to work fine. Just had to add a catch-all case in Rails router to get it to play nice.

### CODE REVIEW

**Comment by kaptron** [comment]
TBD: test on Internet Explorer?  it has a fallback to be turned off in IE (v.9 or less) but we haven't actually tested it yet.

File: app/assets/javascripts/AngularConfig.js.coffee [hunk 1]
```diff
@@ -135,6 +135,10 @@ angular.module('dahlia.controllers',[])
           templateUrl: 'income-calculator/templates/pages/summary.html'
     })
   $urlRouterProvider.otherwise('/') # default to welcome screen
+
+  # have to check if browser supports html5mode (http://stackoverflow.com/a/22771095)
+  if !!(window.history && history.pushState)
```
Comment (chrisdolendo) [inline app/assets/javascripts/AngularConfig.js.coffee]: @kaptron what is the significance of window.history and history.pushState?

**Comment by chrisdolendo** [comment]
:+1: looks good just had one quick question

**Comment by chrisdolendo** [comment]
This was accepted by client--- merging in.

**Pull request merged**